modules/dnn/perf/perf_layer.cpp

```diff
@@ -707,4 +707,65 @@ INSTANTIATE_TEST_CASE_P(/**/, Layer_FullyConnected, Combine(
     dnnBackendsAndTargets()
 ));
 
+struct Layer_Softmax : public TestBaseWithParam<tuple<Backend, Target> >
+{
+    void test_layer(const std::vector<int>& shape, int axis)
+    {
+        int backendId = get<0>(GetParam());
+        int targetId = get<1>(GetParam());
+
+        Mat data(shape, CV_32FC1);
+
+        Scalar mean = 0.f;
+        Scalar std = 1.f;
+        randn(data, mean, std);
+
+        Net net;
+        LayerParams lp;
+        lp.type = "Softmax";
+        lp.name = "testLayer";
+        lp.set("axis", axis);
+
+        int id = net.addLayerToPrev(lp.name, lp.type, lp);
+        net.connect(0, 0, id, 0);
+
+        // warmup
+        {
+            std::vector<String> inpNames(1);
+            inpNames[0] = "data";
+            net.setInputsNames(inpNames);
+            net.setInput(data, inpNames[0]);
```

Comment:
Maybe just `net.setInput(data)` without `setInputsNames`? There is only one input in the model